auth/authenticators/basic-authenticator.ts

```diff
@@ -17,43 +17,62 @@
 import extend = require('extend');
 import { computeBasicAuthHeader, validateInput } from '../utils';
 import { Authenticator } from './authenticator';
-import { AuthenticateOptions, AuthenticatorInterface } from './authenticator-interface';
+import { AuthenticateOptions } from './authenticator-interface';
 
+/** Configuration options for basic authentication. */
 export type Options = {
+  /** The username to be used in basic authorization. */
   username: string;
+  /** The password to be used in basic authorization. */
   password: string;
 }
 
-export class BasicAuthenticator extends Authenticator implements AuthenticatorInterface {
```

Comment:
In the typedoc it would show that  the Authenticators implemented the interface twice. Which felt confusing to me. They already implement it through the Authenticator.